src/com/seafile/seadroid2/SSLTrustManager.java

```diff
@@ -225,7 +226,30 @@ public String getCeritificateInfo() throws CertificateParsingException {
                     + " IssuerAlternative: " + cert.getIssuerAlternativeNames()
                     + " NotAfter: " + cert.getNotAfter();
         }
-        
+
+        private void validateHostName(List<X509Certificate> chain) throws CertificateException {
+            X509Certificate cert = chain.get(0);
+            X509HostnameVerifier mHostnameVerifier = new BrowserCompatHostnameVerifier();
+            boolean verified = true;
+            try {
+                mHostnameVerifier.verify(account.server, cert);
+            } catch (SSLException e) {
+                verified = false;
+            }
+
+            if (!verified) {
```

Comment:
If the verification fails, just throw a `CertificateException`, so that `customCheck` would be called and handle the custom certificate check logic.